src/main/java/edu/wisc/my/portlets/feedback/web/FeedbackFormController.java

```diff
@@ -73,15 +73,31 @@ public void setFeedbackSender(FeedbackSender feedbackSender) {
     @Override
     protected void doSubmitAction(Object command) throws Exception {
         Feedback feedback = (Feedback) command;
-        
+
         feedbackSender.send(feedback);
     }
     @Override
     protected Object formBackingObject(PortletRequest request) throws Exception {
-        final Feedback feedback = new Feedback();
+    	
+    	final Feedback feedback = new Feedback();
+    	
+    	final PortletPreferences preferences = request.getPreferences();
+    	
+    	if(preferences !=null){
+    		request.setAttribute("chatLink",(preferences.getValue("chatLink", "/404.html")));
+    		request.setAttribute("callLink",(preferences.getValue("callLink", "/404.html")));
+    		request.setAttribute("howToLink",(preferences.getValue("howToLink", "/404.html")));
+    	}else{
+    		request.setAttribute("chatLink","/404.html");
+    		request.setAttribute("callLink","/404.html");
+    		request.setAttribute("howToLink","/404.html");
```

Comment:
It would be preferable to `null`, and then not display, missing links rather than to on-purpose display a link that we know will `404`. Don't invite the user to fail.